### PR TITLE
Adding docs

### PR DESCRIPTION
**Pull request by aakarshg**
-> Adds a namespace file that can be used to create a namespace (Helpful for upstream kube)
-> Cleaned up docs and re-arranged them
-> Added a guide to install/contribute

### CODE REVIEW

File: CONTRIBUTE.md [hunk 1]
```diff
@@ -1,18 +1,168 @@
...
+This can be done as follows:
+
+```bash
+# sed -i 's|          image: *|          image: quay.io/<username>/benchmark-operator:latest # |' deploy/operator.yaml
+```
+
+You can then redeploy operator
+```bash
+# kubectl delete -f deploy/operator.yaml
+# kubectl apply -f deploy/operator.yaml
```
Comment (dustinblack) [inline CONTRIBUTE.md]: To go along with the suggestion above:

```bash
# kubectl apply -f /my/testing/operator.yaml
```

File: CONTRIBUTE.md [hunk 2]
```diff
@@ -1,18 +1,168 @@
...
+```
+Redefine CRD
+```bash
+# kubectl delete -f deploy/crds/bench_v1alpha1_bench_crd.yaml
+# kubectl delete -f deploy/crds/bench_v1alpha1_bench_crd.yaml
+```
```
Comment (dustinblack) [inline CONTRIBUTE.md]: I'm not sure why this step would ever be needed as part of workload testing.

Comment (dustinblack) [reply CONTRIBUTE.md]: Also, you have delete in both commands

File: CONTRIBUTE.md [hunk 3]
```diff
@@ -1,18 +1,168 @@
...
+Apply a new CR
+```bash
+# kubectl delete -f deploy/crds/bench_v1alpha1_bench_cr.yaml
+# kubectl delete -f deploy/crds/bench_v1alpha1_bench_cr.yaml
```
Comment (dustinblack) [inline CONTRIBUTE.md]:
```bash
# kubectl apply -f deploy/crds/bench_v1alpha1_bench_cr.yaml
```

File: CONTRIBUTE.md [hunk 4]
```diff
@@ -1,18 +1,168 @@
...
+```
+
+## CI
+Currently we've a CI that runs against PRs.
```
Comment (dustinblack) [inline CONTRIBUTE.md]: we've > we have (more common grammar)

File: CONTRIBUTE.md [hunk 5]
```diff
@@ -1,18 +1,168 @@
...
+If there are mutliple commits, please rebase/squash multiple commits
+before creating the PR by following:
+
+```bash
+$ git checkout <my-working-branch>
+$ git rebase -i HEAD~<num_of_commits_to_merge>
+   -OR-
+$ git rebase -i <commit_id_of_first_change_commit>
+```
+In the interactive rebase screen, set the first commit to `pick` and all others to `squash` (or whatever else you may need to do; RTFM).
```
Comment (jtaleric) [inline CONTRIBUTE.md]: s/; RTFM//

File: CONTRIBUTE.md [hunk 6]
```diff
@@ -1,18 +1,168 @@
...
+### The operator container image
+Any changes to the [roles](roles/) tree or to the [playbook](playbook.yml) file necessitates a new build of the operator container image.
+The container is built using the [Operator SDK](https://github.com/operator-framework/operator-sdk) and pushed to a public repository.
+The public repository could be quay in which case you'll need to:
+
+```bash
+$ operator-sdk build quay.io/<username>/benchmark-operator:testing
+$ docker push quay.io/<username>/benchmark-operator:testing
+```
+
```
Comment (jtaleric) [inline CONTRIBUTE.md]: Add what `:testing` means. Just for the sake of mentioning the tags a user can define for a image.

File: CONTRIBUTE.md [hunk 7]
```diff
@@ -1,18 +1,168 @@
...
+This can be done as follows:
+
+```bash
+# sed -i 's|          image: *|          image: quay.io/<username>/benchmark-operator:latest # |' deploy/operator.yaml
+```
+
+You can then redeploy operator
+```bash
+# kubectl delete -f deploy/operator.yaml
```
Comment (jtaleric) [inline CONTRIBUTE.md]: Possibly delete the previously added CR too... Just to clean all things..

File: README.md [hunk 8]
```diff
@@ -1,117 +1,23 @@
...
-| -------- | --------------------| ------------------ |
-| UPerf                | Network Performance | Working            |
-| FIO                  | Storage IO          | Working            |
-| Couchbase + YCSB     | Databse Performance | WIP            |
+## Workloads status
 
+| Workload                   | Use                  | Status in Operator |
+| -------------------------- | -------------------- | ------------------ |
+| UPerf                      | Network Performance  | Working            |
+| FIO                        | Storage IO           | Working            |
```
Comment (jtaleric) [inline README.md]: Change to FIO-lite

File: docs/installation.md [hunk 9]
```diff
@@ -0,0 +1,79 @@
+## Installation
+This guide uses minishift version v1.33.0+ as the local Kubernetes cluster
+and quay.io for the public registry. We also test on minikube version v0.35+.
+`kubectl` is used but can be substituted* with `oc`.
+
+### Supported versions
```
Comment (jtaleric) [inline docs/installation.md]: Do we want to have a table here instead? 

| Version | Workload | Supported|
|-----------|--------------|---------------|
| 3.11      | Couchbase | |
| 3.11      | UPerf | x |

For example.

File: docs/installation.md [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+* [dep](https://golang.github.io/dep/docs/installation.html)
+* [operator-sdk](https://github.com/operator-framework/operator-sdk)
+* [go](https://golang.org/dl/)
+* [OLM](https://github.com/operator-framework/operator-lifecycle-manager)
+
+The workloads could also have their own requirements which would be specified
+in the installation guide.
+
+### Deploying operator
+Note: Ideally you'd want to clone to your go path
```
Comment (jtaleric) [inline docs/installation.md]: hm, I don't think this is necessary, as long as they have their gopath setup properly for the operator-sdk, no need to build / write code in their $GOPATH.

File: docs/installation.md (hunk 9 above)
Comment (jtaleric) [reply docs/installation.md]: ah, think I miss understood this section.

File: CONTRIBUTE.md [hunk 11]
```diff
@@ -1,18 +1,169 @@
...
+```bash
+# kubectl apply -f deploy/crds/bench_v1alpha1_bench_cr.yaml
+```
+
+## CI
+Currently we have a CI that runs against PRs.
+You can learn more about CI at [work_in_progress]
+
+### CI add test
+To ensure that adding new a workload will not break other workloads and it's
```
Comment (smalleni) [inline CONTRIBUTE.md]: Should be its not it's

File: CONTRIBUTE.md [hunk 12]
```diff
@@ -1,18 +1,169 @@
...
+- Simply copy an existing role and edit as needed
+- Run the `ansible-galaxy init <role_name>` command
+- Pull a blank template from [ansible-roles](https://github.com/ansible-roles/ansible-role-template)
+
+Review the Ansible [role documentation](https://docs.ansible.com/ansible/latest/user_guide/playbooks_reuse_roles.html)
+
+Tasks in the `roles/<role_name>/tasks/main.yml` playbook will be executed when the particular role is triggered in the operator from the Custom Resource (CR).
+
+### Including new roles in the operator
+A new roles should be included in the [playbook](playbook.yml) with
```
Comment (dustinblack) [inline CONTRIBUTE.md]: roles > role (grammar)

File: CONTRIBUTE.md [hunk 13]
```diff
@@ -1,18 +1,169 @@
...
+condition (s) that will [trigger](#Workload-triggers) the role as follows:
```
Comment (dustinblack) [inline CONTRIBUTE.md]: condition (s) > condition(s)   (syntax)

File: CONTRIBUTE.md [hunk 14]
```diff
@@ -1,18 +1,169 @@
...
+### Best practices for new workloads
+The following steps are suggested for your workload to be added:
+* The new role destroys any additional resources it created as part of role in
+case of failure or when disabled. This ensures no interference with subsequent workloads.
+* Please mention any additional cleanup required in your workload guide.
+
+## Testing your workload locally
+
+### The operator container image
+Any changes to the [roles](roles/) tree or to the [playbook](playbook.yml) file will necessitates a new build of the operator container image.
```
Comment (dustinblack) [inline CONTRIBUTE.md]: necessitates > necessitate (grammar)

**Pull request merged**